subsys/mgmt/mcumgr/Kconfig

```diff
@@ -123,6 +123,14 @@ config FS_MGMT_PATH_SIZE
 	  and download commands.
 endif
 
+# Must use the dummy backend for the shell so that mcumgr can capture output
+config MCUMGR_CMD_SHELL_MGMT
+	bool "Enable mcumgr handlers for shell management"
+	depends on !SHELL_BACKEND_SERIAL && SHELL
+	select SHELL_BACKEND_DUMMY
+	help
+	  Enables mcumgr handlers for shell management
```

Comment:
That is a good idea. I will update this later today.

Reply:
I am actually thinking about this more, and I realize that the dummy and serial shell backends don't really have a problem co-existing from what I can see, so long as I am explicit in getting the correct backend pointer when accessing the shell from within mcumgr. 

This means that the serial shell can exist, it can even be used as the backend for mcumgr transport I think (however this would be a bit of a strange use case), and both backends can be used independently without unexpected interactions. 

I think it might make sense to just use the dummy output, and allow the serial one to exist (given that I don't touch the serial backend within mcumgr). I will do some tests on this to be sure I am not missing something.

Reply:
@de-nordic  sorry for the back and forth, but I think this new implementation which doesn't dictate the shell configuration works better. Serial backend is again allowed, but it won't interact with (is independent of) mcumgr which still uses the dummy backend.

Reply:
> Serial backend is again allowed, but it won't interact with (is independent of) mcumgr which still uses the dummy backend.

Right, I agree, I have tested the feature again with UART shell and Bluetooth. Everything works fine and there is no interference.